src/types/api.js

```diff
@@ -5,6 +5,25 @@
  * https://h.readthedocs.io/en/latest/api-reference/
  */
 
+/**
+ * @typedef TextQuoteSelector
+ * @prop {'TextQuoteSelector'} type
+ * @prop {string} exact
+ * @prop {string} [prefix]
+ * @prop {string} [suffix]
+ */
+
+/**
+ * @typedef TextPositionSelector
+ * @prop {'TextPositionSelector'} type
+ * @prop {number} start
+ * @prop {number} end
+ */
+
+/**
+ * @typedef {TextQuoteSelector | TextPositionSelector} Selector
```

Comment:
There is one other type of selector that is actively used by the client, an XPath "range" selector. We can add that later.

Reply:
yep, I saw that! adding that in a follow up PR